### PR TITLE
Make flutter analyze useful for package conflicts

### PR DESCRIPTION
**Pull request by Hixie**
Also, resolve our package conflict, since reflectable has been fixed.

### CODE REVIEW

**Comment by Hixie** [comment]
cc @devoncarew

File: packages/flutter_tools/lib/src/commands/analyze.dart [hunk 1]
```diff
@@ -263,27 +247,19 @@ class AnalyzeCommand extends FlutterCommand {
...
     }
-    if (hadInconsistentRequirements) {
-      if (foundAnyInFlutterRepo)
-        printError('You may need to run "flutter update-packages --upgrade".');
-      if (foundAnyInCurrentDirectory)
-        printError('You may need to run "pub upgrade".');
+
+    // prepare a union of all the .packages files
+    if (dependencies.hasConflicts) {
+      printError(dependencies.generateConflictReport());
```
Comment (devoncarew) [inline packages/flutter_tools/lib/src/commands/analyze.dart]: cool!

**Comment by Hixie** [comment]
Fixes https://github.com/flutter/flutter/issues/2553

**Comment by Hixie** [comment]
I wish the conflict report was actually actionable. As it is, you still have to go and walk through "pub deps --style=list" and "pub upgrade --verbosity=solver" logs to actually figure out what's going on. But at least it gives you a place to start.

File: packages/flutter_tools/lib/src/commands/analyze.dart [hunk 2]
```diff
@@ -589,6 +565,69 @@ linter:
   }
 }
 
+class PackageDependency {
+  Map<String, List<String>> values = <String, List<String>>{};
```
Comment (devoncarew) [inline packages/flutter_tools/lib/src/commands/analyze.dart]: This is a map from package paths, to uses of that package (pubspecs or .packages)? Perhaps a comment to stay that.

**Comment by Hixie** [comment]
Fixes https://github.com/flutter/flutter/issues/2485

**Comment by devoncarew** [comment]
lgtm

**Comment by Hixie** [comment]
Fixes https://github.com/flutter/flutter/issues/1489

File: packages/flutter_tools/lib/src/commands/analyze.dart (hunk 2 above)
Comment (Hixie) [inline packages/flutter_tools/lib/src/commands/analyze.dart]: done.

**Comment by Hixie** [comment]
Oh hey, I forgot to use `canonicalSource`.

**Comment by Hixie** [comment]
@devoncarew I added a second commit that ties a couple of loose ends I forgot about and adds the comment you suggested. PTAL?

**Comment by devoncarew** [comment]
lgtm

**Pull request merged**